app/api/server/v1/teams.ts

```diff
@@ -71,6 +71,44 @@ API.v1.addRoute('teams.create', { authRequired: true }, {
 	},
 });
 
+API.v1.addRoute('teams.convertToChannel', { authRequired: true }, {
+	post() {
+		check(this.bodyParams, Match.ObjectIncluding({
+			teamId: Match.Maybe(String),
+			teamName: Match.Maybe(String),
+			roomsToRemove: Match.Maybe([String]),
+		}));
+		const { roomsToRemove, teamId, teamName } = this.bodyParams;
+
+		if (!teamId && !teamName) {
+			return API.v1.failure('missing-teamId-or-teamName');
+		}
+
+		const team = teamId ? Promise.await(Team.getOneById(teamId)) : Promise.await(Team.getOneByName(teamName));
+		if (!team) {
+			return API.v1.failure('team-does-not-exist');
+		}
+
+		if (!hasPermission(this.userId, 'convert-team', team.roomId)) {
+			return API.v1.unauthorized();
+		}
+
+		const rooms: string[] = Promise.await(Team.getMatchingTeamRooms(team._id, roomsToRemove));
+
+		if (rooms.length) {
+			rooms.forEach((room) => {
+				Meteor.call('eraseRoom', room);
+			});
+		}
+
+		Promise.await(Team.unsetTeamIdOfRooms(team._id));
```

Comment:
Do you need these ones to be sequential, or can we do a `promise.all` with them? 👀